Python/optimizer.c

```diff
@@ -1258,6 +1294,48 @@ int effective_trace_length(_PyUOpInstruction *buffer, int length)
 }
 #endif
 
+static int
+stack_allocate(_PyUOpInstruction *buffer, int length)
+{
+    for (int i = length-1; i >= 0; i--) {
```

Comment:
To my understanding, this is due to the possibility of needing to insert a spill between every instruction right, so you need to reserve 2N number of instructions?